### PR TITLE
Adding Jib based Java building.

### PR DESCRIPTION
**Pull request by patflynn**
Uses [JIB](https://github.com/GoogleContainerTools/jib) as the default skaffold builder for Java

Hi there! I'm the TL for Cloud Code and we'd prefer to use Jib as the Java builder. Jib builds without the need for local Docker or a project Dockerfile, and considerably improves build speed.  

I've included a comment for those who want to use the included Dockerfile.

### CODE REVIEW

**Comment by patflynn** [comment]
@loosebazooka FYI

**Comment by patflynn** [comment]
FYI I'm going to update this to use skaffold profiles to pick build strategy

File: java/java-hello-world/pom.xml [hunk 1]
```diff
@@ -46,6 +46,12 @@
 	</dependencies>
 	<build>
 		<plugins>
+
```
Comment (simonz130) [inline java/java-hello-world/pom.xml]: nit - remove empty line

File: java/java-hello-world/skaffold.yaml [hunk 2]
```diff
@@ -5,8 +5,13 @@ build:
     sha256: {}
   # defines where to find the code at build time and where to push the resulting image
   artifacts:
-  - image: java-hello-world
-    context: .
+    - image: java-hello-world
+      context: .
+      # Comment out the jibMaven line below to use the Dockerfile based build.
+      # To learn more about how jib build Java containers visit
```
Comment (simonz130) [inline java/java-hello-world/skaffold.yaml]: nit: jib builds?

File: java/java-hello-world/skaffold.yaml [hunk 3]
```diff
@@ -5,8 +5,13 @@ build:
...
+      # https://github.com/GoogleContainerTools/jib
+      jibMaven: {}
```
Comment (simonz130) [inline java/java-hello-world/skaffold.yaml]: Will Cloud Code download Jib on the machines where it's not installed? Will it work both on Windows, macOS and Linux the same?

Comment (loosebazooka) [reply java/java-hello-world/skaffold.yaml]: yes, jib is just a maven/gradle plugin, those systems handle downloading everything required.

**Comment by loosebazooka** [comment]
Should we include a mvn wrapper for the maven dependency?

**Comment by etanshaul** [comment]
> Should we include a mvn wrapper for the maven dependency?

I definitely think so. But sounds like a separate PR?

File: java/java-hello-world/skaffold.yaml (hunk 2 above)
Comment (patflynn) [reply java/java-hello-world/skaffold.yaml]: fixed

**Comment by patflynn** [comment]
updated guestbook app and added dockerfile profiles

**Comment by patflynn** [comment]
@simonz130 I just looked at the cloud code docs and they don't seem to go into how the containers are built. Should be good to make this change without any update there.

**Comment by patflynn** [comment]
thanks @simonz130 !

**Pull request merged**